yt/frontends/boxlib/data_structures.py

```diff
@@ -1305,7 +1305,7 @@ def _guess_pcast(vals):
 
 
 def _read_raw_field_names(raw_file):
-    header_files = glob.glob(raw_file + "*_H")
+    header_files = glob.glob(os.path.join(raw_file, "*_H"))
```

Comment:
I don't think this is necessarily correct, is it?  It *may* be correct, but I think we need to verify first that `*_H` is not a suffix applied to files, since we don't use a path separator in the old version.